File: tournaments/reduceString/reduceString.js
function reduceString(inputString) {
    const s = [...inputString];
    let a = s[0];
    let b = s[s.length-1];

    while (1 <= s.length && a === b) {
        s.shift();
        s.pop();
        a = s[0];
        b = s[s.length-1];
    }

    return s.join('');
}

File: tournaments/reduceString/reduceString.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub reduceString {
    my ($inputString) = @_;

    my @store = split('', $inputString);
    my $c = $store[0];
    my $d = $store[@store - 1];

    while (1 <= @store && $c eq $d) {
        shift(@store);
        pop(@store);
        $c = $store[0];
        $d = $store[@store - 1];
    }

    return join('', @store);
}
